app/src/main/java/com/codingblocks/cbonlineapp/fragments/VideoNotesFragment.kt

```diff
@@ -57,77 +48,29 @@ class VideoNotesFragment : Fragment(), AnkoLogger {
                 } catch (cce: ClassCastException) {
                 }
             }
-        })
-        view.notesRv.layoutManager = LinearLayoutManager(context)
+        }, viewModel.notesDao, viewModel.contentDao)
+        view.notesRv.layoutManager = LinearLayoutManager(requireContext())
         view.notesRv.adapter = notesAdapter
-        val itemDecorator = DividerItemDecoration(context!!, DividerItemDecoration.VERTICAL)
-        itemDecorator.setDrawable(ContextCompat.getDrawable(context!!, R.drawable.divider)!!)
+        val itemDecorator = DividerItemDecoration(requireContext(), DividerItemDecoration.VERTICAL)
+        itemDecorator.setDrawable(requireContext().resources.getDrawable(R.drawable.divider))
         view.notesRv.addItemDecoration(itemDecorator)
 
-        notesDao.getNotes(param1!!).observer(this) {
+        viewModel.notesDao.getNotes(param1.toString()).observer(this) {
```

Comment:
it should access it livedata variable and not notesDao 
If there are viewModels for fragment/activity use livedata or mutables variables dao  shouldn't be accessible to fragments/activites it should be a private member of viewmodel

Reply:
Got it